scipy/stats/_axis_nan_policy.py

```diff
@@ -509,7 +557,10 @@ def hypotest_fun(x):
 
             x = np.moveaxis(x, axis, -1)
             res = np.apply_along_axis(hypotest_fun, axis=-1, arr=x)
-            return result_object(*result_unpacker(res))
+            if n_outputs > 1:
+                res = result_unpacker(res)
```

Comment:
```suggestion
            res = result_unpacker(res)
```
This ensures that `res` is a sequence of outputs